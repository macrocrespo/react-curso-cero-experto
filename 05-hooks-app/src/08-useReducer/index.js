export * from './TodoAdd';
export * from './TodoList';
export * from './todoReducer';