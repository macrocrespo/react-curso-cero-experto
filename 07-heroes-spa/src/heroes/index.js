export * from './components';
export * from './helpers';
export * from './hooks';
export * from './pages';
export * from './routes';
export * from './data';