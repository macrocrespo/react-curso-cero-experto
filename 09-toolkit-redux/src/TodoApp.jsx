import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/apis"

export const TodoApp = () => {

    // const { data: todos = [], isLoading } = useGetTodosQuery();
    const [todoId, setTodoId] = useState(1);
    const { data: todo, isLoading } = useGetTodoQuery(todoId);

    return (
        <>
            <h1>Todos - RTK Query</h1>
            <hr />
            <h4>isLoading: { isLoading ? 'True' : 'False' }</h4>

            {/* <pre>
                { todos.map( ({id, title, completed}) => (
                    <li key={ id }>
                        <strong>{ completed ? 'DONE' : 'Pending' } </strong>
                        - { title }
                    </li>
                ) ) }
            </pre> */}

            <pre>{ JSON.stringify( todo ) }</pre>

            <ul>
      
            </ul>

            <button
                onClick={() => { setTodoId( todoId - 1 ) }}
                disabled={ todoId <= 1 }
            >
                Previous TODO
            </button>

            <button
                onClick={() => { setTodoId( todoId +1 ) }}
            >
                Next TODO
            </button>
        </>
    )
}
