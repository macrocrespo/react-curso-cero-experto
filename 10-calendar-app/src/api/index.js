export { default as calendarApi } from './caledarApi';

