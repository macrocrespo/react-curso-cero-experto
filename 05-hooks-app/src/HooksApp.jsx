import { CounterApp } from "./01-useState/CounterApp"
import { CounterWithCustomHook } from "./01-useState/CounterWithCustomHook"
import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook"
import { SimpleForm } from "./02-useEffect/simpleForm"
import { MultipleCustomHooksOptimized } from "./03-examples/MultipleCustomHooksOptimized"
import { FocusScreen } from "./04-useRef/FocusScreen"
import { Layout } from "./05-useLayoutEffect/Layout"
import { CallbackHook } from "./06-memos/CallbackHook"
import { MemoHook } from "./06-memos/MemoHook"
import { Memorize } from "./06-memos/Memorize"
import { Padre } from "./07-tarea-memo/Padre"
import './08-useReducer/intro-reducer';
import { TodoApp } from "./08-useReducer/TodoApp"
import { MainApp } from "./09-useContext/MainApp"

export const HooksApp = () => {
  return (
    <>
        <h1>HooksApp</h1>
        <hr />

        <MainApp />
        
        {/* 
        <TodoApp />
        <Padre />
        <CallbackHook />
        <MemoHook />
        <Memorize />
        <Layout />
        <FocusScreen />
        <MultipleCustomHooksOptimized />
        <FormWithCustomHook />
        <SimpleForm />
        <CounterWithCustomHook />
        <CounterApp />
         */}
    </>
  )
}
