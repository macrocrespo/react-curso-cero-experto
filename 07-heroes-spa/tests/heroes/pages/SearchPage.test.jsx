import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes";

const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate,
}));

describe('Pruebas en <SearchPage />', () => { 

    beforeEach(() => jest.clearAllMocks());
   
    test('Debe mostrarse correctamente con los valores por defecto', () => {

        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );

        expect( container ).toMatchSnapshot();

    });

    test('Debe mostrar a Batman y el input con el valor del queryString', () => {

        render(
            <MemoryRouter initialEntries={['search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('batman');

        const img = screen.getByRole('img');
        expect( img.src ).toContain('/assets/heroes/dc-batman.jpg');


        const noResultsContainer = screen.getByLabelText('no-results');
        expect( noResultsContainer.style.display ).toBe('none');
    });

    test('Debe mostrar un error si no se encuentra el hero (batman123)', () => {

        render(
            <MemoryRouter initialEntries={['search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        const noResultsContainer = screen.getByLabelText('no-results');
        expect( noResultsContainer.style.display ).toBe('');

    });

    test('Debe llamar el navigate a la pantalla nueva', () => {

        const inputValue = 'superman';
        render(
            <MemoryRouter initialEntries={['search']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change( input, { target: { name: 'searchText', value: inputValue } } );
        
        const form = screen.getByRole('form');
        fireEvent.submit( form );
        expect( mockUseNavigate ).toHaveBeenCalledWith( `?q=${ inputValue }` );

    });

 });