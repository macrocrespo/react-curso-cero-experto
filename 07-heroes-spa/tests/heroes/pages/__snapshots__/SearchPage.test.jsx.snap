// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> Debe mostrarse correctamente con los valores por defecto 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching...
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control form-control-lg"
          name="searchText"
          placeholder="Search a hero..."
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-3 btn-lg"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="no-results"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No results were found for 
        <b />
        .
      </div>
    </div>
  </div>
</div>
`;
